### PR TITLE
[TensorFlow] Don't add cast for batch norm when type isn't changing

### PR DESCRIPTION
**Pull request by trevor-m**
TensorFlow batch norm op has two type attributes:
`U` which is the type of the parameters scale, offset, mean, and variance.
`T` which is the type of the input and output.

The TF importer currently adds casts the input from `T` type to `U` type so that the computation is done in `U` type, and then casts the output back to `T`. However, it adds these casts even when `U` and `T` are the same. There ends up being many casts from float to float throughout the model which introduces issues for BYOCG.

### CODE REVIEW

**Comment by trevor-m** [comment]
@kevinthesun Could you please review?

**Comment by srkreddy1238** [comment]
Thanks @trevor-m .
